extensions/blocks/contact-form/deprecated.js

```diff
@@ -0,0 +1,51 @@
+/**
+ * External dependencies
+ */
+import { __ } from '@wordpress/i18n';
+import { InnerBlocks } from '@wordpress/block-editor';
+
+const deprecated = [
+	{
+		attributes: {
+			subject: {
+				type: 'string',
+				default: '',
+			},
+			to: {
+				type: 'string',
+				default: '',
+			},
+			submit_button_text: {
+				type: 'string',
+				default: __( 'Submit', 'jetpack' ),
+			},
+			has_form_settings_set: {
+				type: 'string',
+				default: null,
+			},
+		},
+		migrate: attr => {
+			return {
+				submitButtonText: attr.submit_button_text,
+				hasFormSettingsSet: attr.has_form_settings_set,
+				to: attr.to,
+				subject: attr.subject,
+			};
+		},
```

Comment:
The migration here doesn't consider that the button has been replaced by the Button block.
If we try migrating an old version of the Contact Form, we lose the button.

In Revue I've basically [deprecated](https://github.com/Automattic/jetpack/blob/46f37cae12cd00eb72ceb543fc5bc2c0a3346e3c/extensions/blocks/revue/deprecated/v1/index.js) all button-related attributes, reassigned them to the Button block, and returned that as inner block.
It was quicker there because the attributes names were "compatible" from scratch.

Untested code just as an example:
```jsx
function migrate( attributes, innerBlocks ) {
  const newAttributes = omit( attributes, [
    'submitButtonText',
    'backgroundButtonColor',
    'textButtonColor',
    // etc.
  ] );
  const buttonAttributes = {
    text: attributes.submitButtonText || __( 'Submit', 'jetpack' ),
    backgroundColor: attributes.backgroundButtonColor,
    textColor: attributes.textButtonColor,
    // etc.
  };
  
  const newInnerBlocks = innerBlocks.concat(
    createBlock( 'jetpack/button', {
      element: 'button',
      ...buttonAttributes,
    } );
  );

  return [ newAttributes, newInnerBlocks ];
}
```

Reply:
Thanks, this worked great!